core/templates/domain/skill/skill-summary.model.ts

```diff
@@ -33,8 +33,8 @@ export class SkillSummary {
     public description: string,
     public languageCode: string,
     public version: number,
-    public misconceptionCount: number,
-    public workedExamplesCount: number,
+    public misconceptionCount: number | null,
+    public workedExamplesCount: number | null,
```

Comment:
Why can these be `null`?

Reply:
I assigned it to null because a a `skillBackendSummaryDict` object was defined with typed `BackendSummaryDict` and the `misconception_count` property was initialized to `null`. This created an error `type of null is not assignable to type number`
That's why I went to `core/templates/domain/skill/skill-summary.model.ts` and changed the type to `number | null`

Reply:
> Why can these be `null`?

I've seen that I could instead just use the `!` operator where ever a  new instance of the SkillSummary Object is created like in the screenshot below line 48,49

![image](https://user-images.githubusercontent.com/37558983/159720069-0e16fe5c-aa38-4c22-bc3f-6f7bfb7836b1.png)

Reply:
After setting the type of `misconception_count` and  `work_examples_count` properties of the `SkillSummaryBackendDict` interface to  `number | null`, an error was raced in the body of the static function `createFromBackendDict()` which returns a `SkillSummary`. 

`Argument of type 'number | null' is not assignable to parameter of type 'number'.
  Type 'null' is not assignable to type 'number'.ts(2345)`

The arguments, `misconceptionCount` and `workedExamplesCount` are of type `number`  passed to the `constructor` of `SkillSummary`. 

![image](https://user-images.githubusercontent.com/37558983/160015609-75c28433-e802-4709-a612-028fd5cf5c54.png)

But parameters of the new `SkillSummary` created have values `number | null`

![image](https://user-images.githubusercontent.com/37558983/160016265-49adeefa-59da-4873-8c70-c7c6302c3bc8.png)

So I needed to redefine the types of `misconceptionCount` `and workedExamplesCount` to `number | null` to coincide.

Reply:
@vojtechjelinek I would like you to comment and possible guide me on what to do if it is not correct please

Reply:
ditto as above

Reply:
> ditto as above

Okay sir, I'll look at it
Thanks